packages/oscal-react-library/src/components/OSCALCatalogGroups.js

```diff
@@ -81,27 +81,34 @@ function determineControlExists(groups, controlLayers, rootLayer) {
   if (!upperLayer) {
     return null;
   }
-  // Ensure lowest/deepest control exists
+  // Ensure lowest/deepest control exists within controls/groups
   for (let i = 1; i < controlLayers.length && upperLayer; i += 1) {
-    upperLayer = upperLayer?.controls?.find((control) => control.id === controlLayers[i]);
+    upperLayer =
+      upperLayer?.controls?.find((control) => control.id === controlLayers[i]) ||
+      upperLayer?.groups?.find(
+        (group) =>
+          group.id === controlLayers[i] || conformLinkIdText(group.title) === controlLayers[i]
+      );
```

Comment:
Is it possible to split this out a bit? There is a lot happening here. Maybe have like:

```JavaScript
const find_1 = ...
const find_2 = ...

upperLayer = find_1 || find_2
```

Reply:
Definitely better clarity written that way. I don't this would have much impact on the runtime either.